File: problems/HackerRank/2521_exercises/tree/preOrder.c
void preOrder(struct node *root)
{
	if (root == NULL)
	{
		return;
	}
	printf("%d ", root->data);
	if (root->left)
	{
		preOrder(root->left);
	}
	if (root->right)
	{
		preOrder(root->right);
	}
}

File: problems/LeetCode/Easy/preorder.py
class Solution:
    def preorder(self, root: 'Node') -> List[int]:

        queue = []

        def dfs(node):
            if node == None:
                return
            queue.append(node.val)
            for child in node.children:
                dfs(child)
        
        dfs(root)
        return queue